src/lib/DropDown.svelte

```diff
@@ -11,15 +14,54 @@
 
   let activeValue: string;
   let open = false;
+  let ulElement: HTMLUListElement;
+
+  function handleKeyPress(event: KeyboardEvent) {
+    console.log("handlekyePress");
+    if (!open || disabled) return; // Don't handle if closed or disabled
+
+    const pressedLetter = event.key.toLowerCase();
+
+    // Filter options that start with the pressed letter (ignoring case)
+    const filteredOptions = options.filter((o) =>
+      typeof o === "string"
+        ? o.toLowerCase().startsWith(pressedLetter)
+        : o.value.toLowerCase().startsWith(pressedLetter)
+    );
+
+    // If there are filtered options, select the first one
+    let f: string | number | undefined;
+    if (filteredOptions.length > 0) {
+      f = typeof filteredOptions[0] === "string" ? filteredOptions[0] : filteredOptions[0].id;
+    }
+
+    // Find first button with text content starting with letter pressed and scroll it into view.
+    const btns = ulElement?.querySelectorAll("button");
+    for (let i = 0; i < btns.length; i++) {
+      const btn = btns[i];
+      if (btn.textContent?.toLowerCase()?.startsWith(pressedLetter)) {
```

Comment:
Main change, it'll scroll to the first button with text starting with the letterPressed.

Had to change so it would work with the new way the data is passed.

May have to change in the future, I think it'll only work well if the data in the dropdown is sorted alphabetically, which it isn't guaranteed to be (for now it is, with regions).

Reply:
I'm not sure it is actually, I think it's sorted alphabetically by county code, but we display county names

Reply:
@stignarnia Oh that makes sense. I changed the way the data is provided to the dropdown, so guess I had to change this anyways (I've forgot now, but I believe there was no way to get the button by it's full value anymore after my change so I decided to just rely on the first letter).

Reply:
I can't say I understand what you did, but I tested it, and it works. For example, "VA" is correctly sorted among other countries starting with H as the English name is "Holy See".